src/components/UserSelector.tsx

```diff
@@ -1,45 +1,41 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { fetchUsers } from '../features/users/users';
 
 type Props = {
   value: User | null;
   onChange: (user: User) => void;
 };
 
 export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
   value: selectedUser,
   onChange,
 }) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
   const [expanded, setExpanded] = useState(false);
 
+  const users = useAppSelector(state => state.users.users);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(fetchUsers());
+  }, []);
```

Comment:
Consider adding `dispatch` to the dependency array of the `useEffect` hook to ensure that the effect is consistent with React's rules of hooks. Although `dispatch` is stable in Redux, it's a good practice to include it.